### DISCUSSION TITLE
Programmatically launch Windows applications on SOCA windows desktop using Lambda

### DISCUSSION THREAD

**Discussion by anashanb** [awslabs/scale-out-computing-on-aws#125] [General]
I want to programmatically launch a Windows application on a SOCA Windows desktop using a Lambda function. Are there any built-in SOCA APIs available to achieve this task? If not, what alternatives are available?

**Comment by mcrozes** [comment]
Hello @anashanb  -

SOCA does not provide any API for that, as this is already covered by AWS. I would recommend using AWS Systems Manager as described on https://docs.aws.amazon.com/systems-manager/latest/userguide/walkthrough-powershell.html#walkthrough-powershell-run-script

**Reply by anashanb** [reply]
Is it possible to connect to the SOCA EC2 instance using AWS Systems Manager from my Lambda function in a different AWS account? As far as I know, I can only access the SOCA APIs but not the instance directly from the Lambda function. Any suggestions would be appreciated

**Reply by mcrozes** [reply]
Yes you can do that, however please note AWS Systems Manager is an AWS Service and not specific to SOCA APIs.

To do that, and assuming your Lambda is on a different AWS account, you will just have to make sure your AWS API Client use a correct Role with cross-account permissions (or an IAM user from the AWS account where SOCA is running), ensure you are using the correct region and finally query the AWS Systems Manager API from your Lambda as you would normally do (see link above for example).

In this scenario, your Lambda (AWS Account #2) will query AWS Secrets Manager (AWS Account #1 SOCA) which will execute whatever PowerShell commands/scripts you have specified.
